core/Network/TLS/Parameters.hs

```diff
@@ -141,6 +153,58 @@ defaultParamsClient serverName serverId = ClientParams
     , clientEarlyData               = Nothing
     }
 
+-- | Loads the client cert to provide when the server requests client authentication.
+clientParamsSetClientCert :: FilePath -> FilePath -> ClientParams -> IO (Either String ClientParams)
+clientParamsSetClientCert keyFile certificateFile params = do
+    keys <- E.handle (\(_ :: E.IOException) -> pure []) $ readKeyFile keyFile
+    certs <- readCertificates certificateFile
+    case (keys, certs) of
+        ([], []) -> return $ Left $ "Couldn't read key from key file: " ++ keyFile ++ "\nCouldn't read certs from: " ++ certificateFile
+        ([], _) -> return $ Left $ "Couldn't read key from key file: " ++ keyFile
+        (_:_:_, _) -> return $ Left $ "Multiple keys in key file: " ++ keyFile
+        (_, []) -> return $ Left $ "Couldn't read certs from: " ++ certificateFile
+        ([key], _) ->
+            let hooks = clientHooks params in
+            return $ Right params {
+                      clientHooks = hooks
+                          { onCertificateRequest = \_ -> return (Just (CertificateChain certs, key))}
+                      }
+
+readStore :: FilePath -> IO (Either String CertificateStore)
+readStore path = maybe (Left $ "Couldn't read certs from: " ++ path) Right <$> readCertificateStore path
+
+-- | Apply the merge function to the CA store and the certificate store read from the provided path.
+clientParamsUpdateCAFromStore :: (CertificateStore -> CertificateStore -> CertificateStore) -> FilePath -> ClientParams -> IO (Either String ClientParams)
+clientParamsUpdateCAFromStore mergeStores path = clientParamsUpdateCA mergeStores (readStore path)
+
+-- | Modify the CA store with the provided function.
+clientParamsUpdateCA' :: (CertificateStore -> CertificateStore) -> ClientParams -> ClientParams
+clientParamsUpdateCA' updateStore params =
+      params {
+        clientShared = oldShared { sharedCAStore = store }
+      }
+    where
+      oldShared = clientShared params
+      store = updateStore $ sharedCAStore oldShared
+
+-- | Apply the merge function to the CA store and the certificate store obtained from the action.
+clientParamsUpdateCA :: (CertificateStore -> CertificateStore -> CertificateStore) -> IO (Either String CertificateStore) -> ClientParams -> IO (Either String ClientParams)
+clientParamsUpdateCA mergeStores storeReader params = runExceptT $ do
+    customStore <- ExceptT storeReader
+    return $ clientParamsUpdateCA' (mergeStores customStore) params
```

Comment:
This could be instead implemented as

```haskell
clientParamsUpdateCA mergeStores storeReader params =
    runExceptT ((ExceptT storeReader) <&> (flip clientParamsUpdateCA' params . mergeStores))
    where
      (<&>) = flip fmap
```

But I think it'd be less readable